### PR TITLE
feat(ui): Add release comparison chart

### PR DESCRIPTION
**Pull request by matejminar**
Adding release comparison charts:
- crash free users
- crash free sessions
- users breakdown
- sessions breakdown

This is still feature flagged.

Screenshot 2021-07-01 at 16 48 54

### CODE REVIEW

File: static/app/utils/sessions.tsx [hunk 1]
```diff
@@ -28,3 +35,62 @@ export function getCrashFreeRate(
     ? null
     : getCrashFreePercent(100 - percent(crashedCount ?? 0, totalCount ?? 0));
 }
+
+export function getCrashFreeSeries(
+  groups: SessionApiResponse['groups'] = [],
+  intervals: SessionApiResponse['intervals'] = [],
+  field: SessionField
+): SeriesDataUnit[] {
```
Comment (scttcper) [inline static/app/utils/sessions.tsx:43]:
```suggestion
): (SeriesDataUnit | null)[] {
```

shouldn't need to cast that null below to any, but we might want to filter it out?

File: static/app/views/releases/detail/overview/releaseComparisonChart/sessionsChart.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      case ReleaseComparisonChartType.USER_COUNT:
+      default:
+        return undefined;
+    }
+  }
+
+  getChart():
+    | React.ComponentType<StackedAreaChart['props']>
+    | React.ComponentType<AreaChart['props']>
+    | React.ComponentType<LineChart['props']> {
```
Comment (scttcper) [inline static/app/views/releases/detail/overview/releaseComparisonChart/sessionsChart.tsx:67]: don't need line chart

**Review by scttcper — APPROVED** [review]
👏 easy to read despite the complex chart component. Minor feedback

File: static/app/views/releases/detail/overview/releaseComparisonChart/sessionsChart.tsx (hunk 2 above)
Comment (matejminar) [reply static/app/views/releases/detail/overview/releaseComparisonChart/sessionsChart.tsx]: good catch

File: static/app/utils/sessions.tsx (hunk 1 above)
Comment (matejminar) [reply static/app/utils/sessions.tsx:45]: I filtered out nulls with lodash's compact because it handles types better than the native filter in this case.

**Pull request merged**